### PR TITLE
Activities

### PR DESCRIPTION
**Pull request by andrej-sajenko**
I changed the report activity a bit.

The import activity consist of tree scopes.

1. [receive] receiving file
2. [store] storing chunks
3. [index] indexing chunks

### CODE REVIEW

**Comment by michel-kraemer** [comment]
Looks good to me. Can you create a matrix of all scopes and activities?

**Comment by michel-kraemer** [comment]
Something like this?

 scope/activity  | importing | deleting
-- | -- | --
store | X | X
index | X

**Comment by andrej-sajenko** [comment]
| scope/activity  |  importing | deleting  |
| ------------- |:-------------:|:-----:|
| receive           | x                    | - |
| store               | x                   |   missing |
| index               | x                   |    x |

**Comment by michel-kraemer** [comment]
The scope names could be a bit misleading. I would expect that `store` was in either the `Store` or the `StoreEndpoint` but not in the `ImporterVerticle`. I'm also not sure about `receive` as this is a verb and the other two are nouns. But I don't have a better idea to be honest. I think we need to discuss this further.

**Comment by andrej-sajenko** [comment]
The names are misleading, that is true.
I had the decision to name it like the class where i send the activity
or
to name it like the operation wich happen with the import.
I splitter the import in the

1. receive file
2. store chunks
3. index chunks
events and named the scope: receive, store and index.

If we take the location where we send the activity, it would be

1. storeEndpoint or http
2. store
3. index

What should "scope" mean?
Should it mean
this is the scope where i got the activity?
or
this is the scope of the current activity?

File: georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java [hunk 1]
```diff
@@ -371,9 +374,10 @@ private void onIndexingStarted(long timestamp, int count) {
     
     if (reportActivities) {
       JsonObject msg = new JsonObject()
-        .put("activity", "indexing")
+        .put("activity", "import")
+        .put("state", "index")
         .put("owner", deploymentID())
-        .put("action", "start")
+        .put("action", "enter")
```
Comment (s0x) [inline georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java]: please add the chunk ids to be indexed (e.g. "chunks")

File: georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java [hunk 2]
```diff
@@ -398,11 +402,17 @@ private void onIndexingFinished(long duration, List<String> correlationIds,
     
     if (reportActivities) {
       JsonObject msg = new JsonObject()
-        .put("activity", "indexing")
+        .put("activity", "import")
+        .put("state", "index")
         .put("owner", deploymentID())
-        .put("action", "stop")
+        .put("action", "leave")
         .put("correlationIds", new JsonArray(correlationIds))
```
Comment (s0x) [inline georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java]: please add the IDs of indexed chunks (e.g. "chunks")

File: georocket-server/src/main/java/io/georocket/ImporterVerticle.java [hunk 3]
```diff
@@ -176,12 +178,14 @@ private void onImportingFinished(String correlationId, String filepath,
...
+          .put("activity", "import")
+          .put("state", "store")
           .put("owner", deploymentID())
-          .put("action", "stop")
+          .put("action", "leave")
           .put("correlationId", correlationId)
           .put("chunkCount", chunkCount)
-          .put("duration", duration);
+          .put("duration", duration)
+          .put("layer", layer);
```
Comment (s0x) [inline georocket-server/src/main/java/io/georocket/ImporterVerticle.java]: Please add the type of the file to the message (e.g. mimetype)

Comment (s0x) [inline georocket-server/src/main/java/io/georocket/ImporterVerticle.java]: You might do the same for onImportingStarted

File: georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java (hunk 1 above)
Comment (andrej-sajenko) [inline georocket-server/src/main/java/io/georocket/index/IndexerVerticle.java]: i add _chunkIds_ as array => chunkCount == chunkIds.size()

**Comment by andrej-sajenko** [comment]
review pls

**Pull request merged**